src/AWS.Deploy.Recipes/DeploymentBundleDefinitions/Container.deploymentbundle

```diff
@@ -11,6 +11,16 @@
             "AdvancedSetting": true,
             "Updatable": true
         },
+        {
+            "Id": "DockerfilePath",
+            "Name": "Dockerfile Path",
+            "Description": "Specifies the absolute path to the Dockerfile as an alternative to the default location.",
+            "Type": "String",
+            "TypeHint": "DockerfilePath",
+            "DefaultValue": "",
+            "AdvancedSetting": true,
+            "Updatable": true
+        },
         {
             "Id": "DockerExecutionDirectory",
```

Comment:
If the `DockerExecutionDirectory` always needs to be the parent directory of `DockerfilePath`, then we shouldn't be exposing it to the user as an option setting. We can set it ourselves based on whatever the `DockerfilePath` is

Reply:
1. Added a recipe-level validator to validate their relationship.
    * I had originally tried to list the validator in `Container.deploymentbundle` as opposed to in every Docker-based recipe, but looks like we're not set up to load validators from the deployment bundle files, only the list of options. Opted to add to the recipes for now, though I imagine we could consolidate in the future.
3. I don't believe it's always the direct parent, per https://docs.docker.com/engine/reference/commandline/build/#text-files _"The path must be to a file within the build context."_

Contrived example:
```
sln
    csproj <- we may want to build here
        /BuildTools
            /Docker
                /Dev
                    Dockerfile <- but this can be in any child
                /Prod
```

Reply:
We don't necessarily know what the `DockerExecutionDirectory` is. We still need to allow users to set it.

Reply:
Added a recipe-level `DockerfilePathValidator` to check the relationship between the Dockerfile path and the Docker execution directory.